handlers/collectionID/handler.go

```diff
@@ -0,0 +1,16 @@
+package collectionID
+
+import (
+	"net/http"
+	"context"
+	"github.com/ONSdigital/go-ns/common"
+)
+
+// Handler is a wrapper which adds a CollectionID header if one does not yet exist
+func Handler(h http.Handler) http.Handler {
+
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			collectionID := req.Header.Get(common.CollectionIDHeaderKey)
```

Comment:
Only comment I have is to add an `if present` check. i.e. only add the item to the context if the header exists.